src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -405,13 +427,24 @@ protected IEnumerable<Symbol> GetUnsafeAddressTaken()
 
         #region Tracking reads/writes of variables for warnings
 
-        protected virtual void NoteRead(Symbol variable, ParameterSymbol rangeVariableUnderlyingParameter = null)
+        protected virtual void NoteRead(
+            Symbol variable,
+            ParameterSymbol rangeVariableUnderlyingParameter = null)
         {
             var local = variable as LocalSymbol;
             if ((object)local != null)
             {
-                _usedVariables.Add(local);
+                if (IsCapturedInLocalFunction(local,
+                                              rangeVariableUnderlyingParameter))
+                {
+                    NoteReadInLocalFunction(local, local);
```

Comment:
I see, that is what `IsCapturedInLocalFunction` is intended to check (if we are analyzing a local function and the variable is captured in it)

---
In reply to: [73435782](https://github.com/dotnet/roslyn/pull/12888#discussion_r73435782) [](ancestors = 73435782)